frontend/src/metabase/lib/hooks.js

```diff
@@ -0,0 +1,34 @@
+import { useRef, useEffect, useMemo, useState } from "react";
+
+export function useAsyncFunction(fn) {
```

Comment:
does this hook seem sound? I looked at a few third party alternatives and they all annoyingly expose the term "mounted" which I think is unnecessary.

Reply:
the goal here being to avoid triggering things within a component after it has unmounted. I've wrapped the given `fn` in a `Promise` that won't resolve if the component has unmounted